### PR TITLE
Switching from babylonjs to @babylonjs/core

### PR DESCRIPTION
**Pull request by JuliaABurch**
## Description
This change moves our BabylonJS implementation away from the assumption that BabylonJS will be imported/embedded as an external source file in its entirety. This caused issues for webapps that use bundling to package dependencies together with their code. 

Furthermore, this change switches from using BabylonJS' UMD package to the ES6 modules (`@babylonjs/core`, etc)  instead. This was motivated by the fact that the editor creates projects that use the latter, and subtle bugs would be introduced by mixing the two -- however, this change imposes a restriction on the potential use of this library, as detailed below.

Code changes largely involved explicitly importing required BabylonJS classes, instead of assuming `BABYLON` global would be defined. Although there's a way to import `BABYLON` from a `@babylonjs/core` legacy module, doing it this way will allow tree-shaking to prune out the parts of BabylonJS our library doesn't use. The API of the two packages are exactly the same, so no implementation details were changed.

The integration tests had their functionality defined in a <script> tag on the html page - these have been moved to a separate .js file that will be compiled by webpack. Again, no implementation details were changed; the .js files are now linted, so minor cosmetic changes were made to address warnings.

I've not touched the `README` as we have a separate task that encompasses all the changes we need to make.

This change does come with a caveat:
Although a user *can* embed `babylonjs.min.js` (the UMD package) through a script tag and continue to use our library with this dependency (as the integration tests did previously, and as our `README` guides people to do), there will nevertheless be subtle bugs introduced from mixing babylonjs and @babylonjs/core. (Animations work, text-to-speech does not.) It *is* valuable to be able to use `babylonjs` this way for rapid prototyping, even as people are being encouraged to move towards @babylonjs/core for implementation of their actual web apps. On the other hand, users who were already using @babylonjs/core would have had to use workarounds to use our library.

## Submission Checklist
I confirm that I have...
- [x] removed hard-coded Cognito IDs
- [x] manually smoke-tested the BabylonJS integration tests
- [x] manually smoke-tested the BabylonJS demos
- [x] manually smoke-tested the Three.js integration tests
- [x] manually smoke-tested the Three.js demo

---
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: packages/amazon-sumerian-hosts-babylon/package.json [hunk 1]
```diff
@@ -24,6 +24,7 @@
     "aws-sdk": "^2.1094.0"
   },
   "peerDependencies": {
-    "babylonjs": "=4.2.1"
+    "@babylonjs/core": "=4.2.1",
```
Comment (prestomation) [inline packages/amazon-sumerian-hosts-babylon/package.json:27]: do we want `=`? Should we use `^` so naturally test with newer versions?
Also..babylon is on version 5. we should be using that now right? are we tracking this?

**Review by prestomation — DISMISSED** [review]

**Review by eherozhao — APPROVED** [review]

File: packages/amazon-sumerian-hosts-babylon/src/Babylon.js/HostObject.js [hunk 2]
```diff
@@ -5,6 +5,11 @@ import {
   LipsyncFeature,
   GestureFeature,
 } from '@amazon-sumerian-hosts/core';
+import {SceneLoader} from '@babylonjs/core/Loading/sceneLoader';
```
[the remark this answers is not in the document]
Comment (JuliaABurch) [reply packages/amazon-sumerian-hosts-babylon/src/Babylon.js/HostObject.js:8]: Doing these very explicit imports is actually how we take advantage of tree-shaking! (Additionally, it's not possible to `import { ... } from '@babylonjs/core'`)

File: packages/amazon-sumerian-hosts-babylon/package.json (hunk 1 above)
Comment (JuliaABurch) [reply packages/amazon-sumerian-hosts-babylon/package.json:27]: `4.2.1` is the latest/last version of 4.x before 5.x; `4.2.0` has a bug with .glb loaders, thus my decision not to use `<=`

BabylonJS 5.x won't be officially released until May -- and introduces some breaking changes, so we'd need to update our library to work with it. (I spent some time testing our repo with 5.2.0) The BabylonJS Editor targets 4.2.1, thus my "locking" it to this change

Comment (prestomation) [reply packages/amazon-sumerian-hosts-babylon/package.json:27]: ah ok. should we do `^4.2.1` then?

and didn't realize it wasn't out yet. 5 looks to be the latest on NPM so maybe I misunderstand what 'release' means :)

**Review by eherozhao — DISMISSED** [review]

File: packages/amazon-sumerian-hosts-babylon/package.json (hunk 1 above)
Comment (JuliaABurch) [reply packages/amazon-sumerian-hosts-babylon/package.json]: Haha, yeah - I mispoke a bit - [the forums banner](https://forum.babylonjs.com/) just says they won't be acknowledging its release until May.

**Review by eherozhao — APPROVED** [review]

**Review by prestomation — APPROVED** [review]

**Pull request merged**